lib/iris/coord_systems.py

```diff
@@ -791,6 +791,15 @@ def __repr__(self):
                    self.false_easting, self.false_northing,
                    self.secant_latitudes, self.ellipsoid)
 
+    def __eq__(self, other):
+        sattrs = self.__dict__.copy()
+        oattrs = other.__dict__.copy()
+        s_secant_latitudes = sattrs.pop('secant_latitudes', None)
+        o_secant_latitudes = oattrs.pop('secant_latitudes', None)
+        return (self.__class__ == other.__class__ and
+                sattrs == oattrs and
+                (s_secant_latitudes==o_secant_latitudes).all())
```

Comment:
Having looked at the test failures, sometimes `secant_latitudes` is an array, sometimes it's a tuple or list. You could extend the comparison even further so it can cope with either, but it would be easier if the class always ensured that the attribute was one type. Personally I'd go with ensuring it's always a tuple.